code/ai/aicode.cpp

```diff
@@ -11983,7 +11983,7 @@ void ai_do_repair_frame(object *objp, ai_info *aip, float frametime)
 {
 	static bool rearm_eta_found=false;
 
-	if (Ships[objp->instance].team == Iff_traitor) {
+	if ( (Ships[objp->instance].team == Iff_traitor) && ((aip->ai_flags[AI::AI_Flags::Awaiting_repair]) || (aip->ai_flags[AI::AI_Flags::Being_repaired])) ) {
```

Comment:
Why not move the traitor check and make it the first statement inside the next `if()` block?

That would also allow the `else` at the end of the function to properly execute.

Reply:
I am honestly not sure any more why I did it exactly this way.  Let me take a look.

Reply:
Ok.  Yes, that's a good idea.